File: BE/Src/MIT.Fwk.Infrastructure/Entities/UserDevice.cs
using JsonApiDotNetCore.Resources;
using JsonApiDotNetCore.Resources.Annotations;
using System;
using System.ComponentModel.DataAnnotations.Schema;

namespace MIT.Fwk.Infrastructure.Entities
{
    [Resource]
    [Table("UserDevices")]
    public class UserDevice : Identifiable<Guid>
    {
        [Attr]
        public string userId { get; set; }

        [Attr]
        public string deviceHash { get; set; }

        [Attr]
        public string salt { get; set; }

        [Attr]
        public DateTimeOffset? lastAccess { get; set; }

        [Attr] public string PushToken { get; set; }
        [Attr] public string AppleToken { get; set; }
        [Attr] public string GoogleToken { get; set; }
        [Attr] public string FacebookToken { get; set; }
        [Attr] public string TwitterToken { get; set; }
        [Attr] public string Platform { get; set; }
        [Attr] public string DeviceName { get; set; }

        [HasOne]
        public virtual User User { get; set; }

    }
}

File: FE/WEB/app/models/user-device.js
import Model, { attr } from '@ember-data/model';

export default class UserDeviceModel extends Model {
  @attr('string', { defaultValue: '' }) userId;
  @attr('string', { defaultValue: '' }) deviceHash;
  @attr('string', { defaultValue: '' }) salt;
  @attr('date', {
    defaultValue() {
      return new Date();
    },
  })
  lastAccess;
  @attr('string', { defaultValue: '' }) pushToken;
  @attr('string', { defaultValue: '' }) appleToken;
  @attr('string', { defaultValue: '' }) googleToken;
  @attr('string', { defaultValue: '' }) facebookToken;
  @attr('string', { defaultValue: '' }) twitterToken;
  @attr('string', { defaultValue: '' }) platform;
  @attr('string', { defaultValue: '' }) deviceName;
}
